specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json

```diff
@@ -1984,6 +2272,88 @@
         }
       }
     },
+    "SqlServerInstanceProperties": {
+      "description": "Properties of SqlServerInstance.",
+      "type": "object",
+      "properties": {
+        "version": {
```

Comment:
>version" [](start = 9, length = 8)

what properties are required on a create?  Are any properties read only?  Please mark them as such

Reply:
All properties are needed for creating a new resource of this resource type. If later any of them can be optional, will update the spec.

Reply:
They should be marked as required then

Reply:
Will mark them as "required"

Reply:
Sorry, required is a string array at the level above the properties.  See how "name" is marked required in the "Sku" definition in this file as an example